R/ptlm.R

```diff
@@ -102,22 +102,35 @@ plc_50 <- function(p_abs, pah = NULL, NLC50 = NULL) {
   NLC50 / (1 + p_abs^TLM_a/TLM_R)
 }
 
-#' Look up the NLC50 value for a PAH.
+#' Calculate the NLC50 value for a PAH using the Target Lipid Model (TLM)
+#'
+#' This uses the equation and default values from McGrath et al. 2018.
 #'
 #' @param pah The PAH of interest
+#' @param slope Slope in Equation 1 in McGrath et al. 2018. Default `-0.94`.
+#' @param HC5 The critical target lipid body burden above which 95% of species
+#'    should be protected in μmol/g octanol. Default `9.3`.
 #'
 #' @return NLC50 value, in ug/L
 #' @export
+#' @references
+#'  McGrath, J.A., Fanelli, C.J., Di Toro, D.M., Parkerton, T.F., Redman, A.D.,
+#'  Paumen, M.L., Comber, M., Eadsforth, C.V. and den Haan, K. (2018),
+#'  Re-evaluation of target lipid model–derived HC5 predictions for hydrocarbons.
+#'  Environ Toxicol Chem, 37: 1579-1593. https://doi.org/10.1002/etc.4100
```

Comment:
@atillmans this would benefit from your eyes on it. Particularly lines 110-112 where I describe the slope and HC5 parameters.

Reply:
Suggest replacing line 107 with:
This uses the equation and calculated values from McGrath et al. 2018

Suggest replacing lines 110-112 with the following:
@param slope Slope taken from Table 3 in McGrath et al. 2018  Calculated value: -0.94
@param HC5 is the 5th percentile of the SSD of critical body burdens predicted to be hazardous for no more than 5% of the species.  Calculated value: 9.3 umol/g

Reply:
Thanks @atillmanns. How about something like this for the parameter descriptions, where we describe _what_ the parameter is, then state the default (and its origin):

```
@param slope The Slope in Equation 1 in McGrath et al. 2018. The default value is -0.94, which is taken from Table 3 in McGrath et al. 2018. It is not recommended to adjust this without good justification.
@param HC5 The 5th percentile of the SSD of critical body burdens predicted to be hazardous for no more than 5% of the species. Default value is 9.3 umol/g, which was calculated using Equation 3 in McGrath et al 2018. It is not recommended to adjust this without good justification.
```

Reply:
Hi @ateucher yes that looks fine.  Also, the chemical class coefficients (delta c) are specific to the TLM so these need to be included in the default values.  As new iterations of the TLM are published, we will update these parameters.

Reply:
Also, they are not default values.  They are coefficients from the TLM calculated in McGrath et al. 2018.  That's why I replaced default with calculated.  Perhaps "model coefficient" is a better phrase.

Reply:
> the chemical class coefficients (delta c) are specific to the TLM so these need to be included in the default values. As new iterations of the TLM are published, we will update these parameters.

Will the delta c always be the same for all PAHs?

> Also, they are not default values.

They are default values in the context of running the function - i.e., a user can run the `nlc50()` function without explicitly specifying the slope and HC5, and the values `slope = -0.94, HC5 = 9.3` will be used _by default_.

E.g., `nlc50("Benzo[a]pyrene")` is equivalent to `nlc50("Benzo[a]pyrene", slope = -0.94, HC5 = 9.3)`. 

As new iterations of the TLM are published, you can run the `nlc50()` function and specify the new values; eg: `nlc50("Benzo[a]pyrene", slope = -0.89, HC5 = 9.6)`

Reply:
We could change the language to something like "By default, the value -0.94 is used, as calculated in Mcgrath 2018..."?

Reply:
I see, yes default values in the program.  Makes sense.  I think we can leave it as is.  As for delta c, it is also an output of the TLM so it will change with new iterations of the TLM.  Also, if our paper is accepted, there will be a delta c for PAHs and for heterocyclic aromatic hydrocarbons (HACs).  It would be great if we could add the capacity to calculate the PTL50 for both these groups in the future.

Reply:
Ok great, so to retain the flexibility for PAHs and HACs, the function signature should 
look like:

```r
nlc50(chemical, slope = -0.89, HC5 = 9.6, dc_pah = -0.364, dc_hac = -0.471)
```

Then the function will look up the chemical class (PAH/HAC), the log(Kow), and the molecular weight based on the chemical name, and use the values provided in the function call (with default values as above, which can be overridden by the user)?